packages/commonwealth/client/scripts/state/api/threads/getThreadsById.ts

```diff
@@ -1,50 +1,28 @@
-import { useQuery } from '@tanstack/react-query';
-import axios from 'axios';
-import Thread from 'models/Thread';
-import { ApiEndpoints, SERVER_URL } from 'state/api/config';
+import { trpc } from 'client/scripts/utils/trpcClient';
 
 const THREAD_STALE_TIME = 5000; // 5 seconds
 
 interface GetThreadsByIdProps {
-  communityId: string;
-  ids: number[];
+  community_id: string;
+  thread_ids: number[];
   apiCallEnabled?: boolean;
 }
 
-const getThreadsById = async ({
-  communityId,
-  ids,
-}: GetThreadsByIdProps): Promise<Thread[]> => {
-  const response = await axios.get(
-    `${SERVER_URL}${ApiEndpoints.FETCH_THREADS}`,
-    {
-      params: {
-        community_id: communityId,
-        thread_ids: ids,
-      },
-    },
-  );
-
-  return response.data.result.map((t) => new Thread(t));
-};
-
 const useGetThreadsByIdQuery = ({
```

Comment:
I wonder if switching to tRPC here breaks the `cacheUpdater` function in `packages/commonwealth/client/scripts/state/api/threads/helpers/cache.ts` since `ApiEndpoints.FETCH_THREADS` no longer applies to this query.

Reply:
let's discuss this on Friday, this cache updater smells like accidental complexity

Reply:
It would break if we move older thread API endpoints to newer trpc routes (haven't tested the pr yet). 

Originally when this file was added, we had many different get thread APIs (at least 3 that I remember) with non-standard responses, but the APIs got standard responses overtime. 

There were 2 possibilities when user mutated any thread data
1. Invalidate all thread react query cache to fetch fresh data.
2. Update the react query cache and only that thread which got mutated. 

1 was getting annoying if the user had visited 5 different topics, scrolled a bit to load 100 threads/topic, and was constantly mutating thread objects, we were basically re-loading threads list with every mutate action (tons of API calls), so we went with 2 (I know its too complex/old now, but again at that time it simplified stuff).

Now since most thread APIs have standard responses, we could create a new cacher updater with trpc and simplify all the complex stuff. Or try `1` but I don't recommend it mainly because I think a serious user would get annoyed by so many data loads b/w mutations.

cc: @Rotorsoft @timolegros

Reply:
If you look at the latest code, you'll find that I'm trying to unify all thread query outputs to ThreadView. Once we have a full list of queries, we can think about consolidation (less queries with more options) and simplification of the caching layer... for this particular PR all we need is to verify that the caching keeps working

Reply:
Sounds good, Ty!